### ISSUE TITLE
Button component issues

### ISSUE THREAD

**Issue by jacecotton** [jacecotton/tcds#20]
Several issues with the current [tcds-button](https://github.com/jacecotton/tcds/tree/main/components/button) component.

The `controls` prop is invalid - IDREFs do not cross shadow boundaries (see #18). `aria-controls` should be placed directly on the host element.

Problem with *that* is `aria-controls` should be placed on the element being clicked *on*, not *through*. The element must have `role=button`, etc.

So, the host element itself needs to be the button, not a container for a button (as is currently the case). Current thought is to instead make the component a customized built-in element, something along the lines of:

```html
<button is="tcds-button" aria-controls="...">...</button>
<a is="tcds-link-button" href="...">...</a>
```

There are pros and cons to this approach.
* Cons:
    * Safari does not support customizing built-ins, but we could add a polyfill (see #19).
    * Link buttons.
        * Whereas the current component automatically decides whether to create an `a` or `button` element based on the presence of the `link` prop, this would now be the responsibility of the component user or CMS.
        * Duplication - `variant`, `icon`, `size`, etc. still applies to both real and link buttons. Hypothetically, they could possibly share a base class that mixes in `HTMLLinkElement` and `HTMLButtonElement`, but executing this may be tricky or even impossible.
    * We cannot attach a shadow root to `button` or `a` elements, so styles would be unscoped and unencapsulated, and we can't use `slot`s.
* Pros:
    * No longer any weirdness with focus and event delegation.
    * Not JavaScript-dependent - unlike most of our other components, `tcds-button` is currently inaccessible and completely non-functional without JavaScript. This violates the progressive enhancement principle.

**Comment by jacecotton** [comment]
Quick draft snippet to explore the solution to the duplication issue:

```js
const ButtonBase = (ElementInterface) => class extends ElementInterface {
  static get observedAttributes() {
    return ["icon", "size", "variant"];
  }

  // if icon only, add aria-label and title
  // insert icon into right spot
};

class Button extends ButtonBase(HTMLButtonElement) {

}

class LinkButton extends ButtonBase(HTMLAnchorElement) {

}

customElements.define("tcds-button", Button, {extends: "button"});
customElements.define("tcds-link-button", LinkButton, {extends: "a"});
```

**Comment by jacecotton** [comment]
If we go with the proposed solution, here is the to-do list of what would need to be done in the existing codebases:

* For each component, replace `<tcds-button>` with appropriate element (either `<a is="tcds-link-button">` or `<button is="tcds-button">`.
* In every theme directory of each site on TCDS, CMD+F for `tcds-button` and `::part(button)`, replace accordingly.
* For Site Studio plugin, use Twig to decide whether to use `<a is="tcds-link-button">` or `<button is="tcds-button">`.
